### ISSUE TITLE
Certain regex pattern in class constant causes PHP class to be ignored

### ISSUE THREAD

**Issue by jaromic** [doxygen/doxygen#6818]
This works as expected:

```
/**
 * ...
 */
class User {
    const TEST = '/[[:alnum:]]/';
    // [...]
}
```

This causes the `User` class to be ignored:

```
/**
 * ...
 */
class User {
    const TEST = '/[[:alnum:]x]/';
    // [...]
}
```

PHP 7.2.4, Doxygen 1.8.15.

**Comment by jaromic** [comment]
My configuration:
[Doxyfile.txt](https://github.com/doxygen/doxygen/files/2827291/Doxyfile.txt)

**Comment by jaromic** [comment]
The classes in this example are different in one blank character only, but `Class1` will be included and `Class2` will not:

[example_doxygen_issue_6818.tar.gz](https://github.com/doxygen/doxygen/files/2832379/example_doxygen_issue_6818.tar.gz)

**Comment by albert-github** [comment]
Some tests:

In the file `Class2.php` I renamed the class from `Class1` to `Class2` so we have distinctive names.
- With version 1.8.15 we only see `Class1`
- With the current development version of doxygen we have an extra problem: we don't see any class.
  - With commit c581ea17bac7dd4ebba1ca53a55ef592f96e0117 from Sat Jan 5 20:23:45 2019 (Fixed alt attribute appearing in \<object\> tag for SVG images) we still see `Class1`
  - With the update c61d8701470352385bcf5720ee516ba95c4393c8 form Mon Jan 7 21:51:07 2019 (issue #6734 parsing performance worsened) `Class1` is gone as well.

**Comment by albert-github** [comment]
I did some further tests and the original problems looks like to come from: d8001efd89146e04d92f5ea41ab27a7de09b6c53
Thursday, November 09, 2017 9:47:21 PM (Bug 787952 - Problem parsing c++ gnu::visibility)
Adding the line:
`if (!insideCpp) REJECT;`
 in `scanner.l` with the condition `<*>"[["`
works (should be tested when the total disappearance of `Class1` and `Class2` is gone.

Elaborating on this for the current version results in:
I've just pushed a proposed patch, pull request #6823

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
